### ISSUE TITLE
Rebind creates new beans, hence loose state

### ISSUE THREAD

**Issue by WonderCsabo** [excilys/androidannotations#1490]
Let's say we have the following bean:
```java
class MyBean {
  @RootContext
  Context context;

  @Bean
  MyOtherBean bean;
}
```

The generated class will have the following `rebind()` method:
```java
void rebind(Context context) {
  context_ = context;
  init_();
}
```
The problems is that `rebind()` just calls the `init_()` method, so a new instance of `MyOtherBean` will be injected. I think rebind should not do that, but retain the same instance for `MyOtherBean` and only reinject the `Context`-related stuff.

@yDelouis @dodgex WDYT?

**Comment by yDelouis** [comment]
`rebind` should call `rebind` on all its `EBean`. I don't know if one may need a `@AfterRebind` method.

**Comment by WonderCsabo** [comment]
@yDelouis i agree, but we still have to modify rebind, because it has to reinject `Contex` stuff. I do not think `@AfterRebind` is needed.

**Comment by yDelouis** [comment]
Yes, we have to make some modifications.
